modules/openapi-generator/src/main/resources/JavaSpring/pathParams.mustache

```diff
@@ -1 +1 @@
-{{#isPathParam}}{{#useBeanValidation}}{{>beanValidationPathParams}}{{/useBeanValidation}}{{#oas3}}@Parameter(name = "{{baseName}}", description ={{/oas3}}{{^oas3}}@ApiParam(value ={{/oas3}} "{{{description}}}"{{#required}}, required = true{{/required}}{{#allowableValues}}, allowableValues = "{{#enumVars}}{{#lambdaRemoveDoubleQuote}}{{{value}}}{{/lambdaRemoveDoubleQuote}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/enumVars}}"{{/allowableValues}}{{^isContainer}}{{#defaultValue}}, defaultValue = "{{{.}}}"{{/defaultValue}}{{/isContainer}}) @PathVariable("{{baseName}}") {{>optionalDataType}} {{paramName}}{{/isPathParam}}
+{{#isPathParam}}{{#useBeanValidation}}{{>beanValidationPathParams}}{{/useBeanValidation}} {{>paramDoc}} @PathVariable("{{baseName}}") {{>optionalDataType}} {{paramName}}{{/isPathParam}}
```

Comment:
Any chance you could also push in here the `{{#isDate}}` and `{{#isDateTime}}` blocks from the `queryParams.mustache` below, as a side-thought?

Reply:
See the last commit. I introduced `dateTimeParam.mustache` and call it from queryParams and pathParams. This is basically the same functionality as #10879 but with a new template to deduplicate the code.

Are there other param types where the DateTimeFormat annotation would make sense? headerParams? formParams?

Do you have any OpenApi specifications to include in the tests?

Reply:
Much appreciated. Minimal yaml for the path variable use case would be:

```yaml
openapi: 3.0.3
info:
  title: Api Documentation
  version: '1.0'
paths:
  /thingy/{date}:
    get:
      operationId: get
      parameters:
        - name: date
          in: path
          required: true
          schema:
            type: string
            format: date
            example: '2021-01-01'
      responses:
        '200':
          description: OK
```

I can imagine this to be useful both for form data and header, but I don't have a use case for either of those.

Reply:
The last commit applies the @DateTimeFormat annotation consistently across the different parameter types. A test configuration is added based on your example spec.